### PR TITLE
Fix ArrayIndexOutOfBoundsException from HdrHistogram

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

DoubleRecorder from HdrHistogram can throw
ArrayIndexOutOfBoundsException even if auto-resize is enabled for its
DoubleHistograms when the range of values is too wide. For example, when
a histogram contains 0.000000001 (1 nanosecond), it can only contain up
to 16, which is way too small for our use case.

Modifications:

- Limit the minimum value to 0.001 (1 millisecond)
- Limit the maximum value to the extent that does not trigger
  ArrayIndexOutOfBoundsException
- Fix an overflow bug in rotate() method due to incorrect cast order

Result:

ArrayIndexOutOfBoundsException should not be raised even if too small or
big value is recorded.

### CODE REVIEW

File: core/src/test/java/com/linecorp/armeria/common/metric/RollingHdrQuantilesTest.java [hunk 1]
```diff
@@ -47,11 +47,43 @@
...
+        for (;;) {
+            final long mid = max / 2 + min / 2;
+            if (min == mid) {
+                return min;
+            }
+
+            final DoubleRecorder recorder = new DoubleRecorder(numberOfSignificantValueDigits);
+            try {
+                recorder.recordValue(RollingHdrQuantiles.MIN_VALUE);
+                recorder.recordValue(mid);
```
Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/common/metric/RollingHdrQuantilesTest.java]: Oops~ OOM~ here~

Comment (hyangtack) [reply core/src/test/java/com/linecorp/armeria/common/metric/RollingHdrQuantilesTest.java]: https://ci.appveyor.com/project/trustin/armeria/build/master.1359
```
com.linecorp.armeria.common.metric.RollingHdrQuantilesTest > valueRanges FAILED
    java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.atomic.AtomicLongArray.<init>(AtomicLongArray.java:81)
        at org.HdrHistogram.ConcurrentHistogram$AtomicLongArrayWithNormalizingOffset.<init>(ConcurrentHistogram.java:609)
        at org.HdrHistogram.ConcurrentHistogram.resize(ConcurrentHistogram.java:355)
        at org.HdrHistogram.DoubleHistogram.handleShiftValuesException(DoubleHistogram.java:560)
        at org.HdrHistogram.DoubleHistogram.shiftCoveredRangeToTheLeft(DoubleHistogram.java:523)
        at org.HdrHistogram.DoubleHistogram.autoAdjustRangeForValueSlowPath(DoubleHistogram.java:408)
        at org.HdrHistogram.DoubleHistogram.autoAdjustRangeForValue(DoubleHistogram.java:379)
        at org.HdrHistogram.DoubleHistogram.recordSingleValue(DoubleHistogram.java:347)
        at org.HdrHistogram.DoubleHistogram.recordValue(DoubleHistogram.java:293)
        at org.HdrHistogram.DoubleRecorder.recordValue(DoubleRecorder.java:76)
        at com.linecorp.armeria.common.metric.RollingHdrQuantilesTest.findMaxValue(RollingHdrQuantilesTest.java:73)
        at com.linecorp.armeria.common.metric.RollingHdrQuantilesTest.valueRanges(RollingHdrQuantilesTest.java:57)
```

**Comment by imasahiro** [comment]
Do we have any chance of releasing new 0.54.x binaries?

**Comment by trustin** [comment]
Working on 0.55. Would 0.55 work for you?

**Comment by imasahiro** [comment]
Depends on how difficult catching up zipkin2 upgrade is since it contains zipkin API changes..

**Comment by trustin** [comment]
Ah, I see. Let me release 0.54.2.

**Pull request merged**